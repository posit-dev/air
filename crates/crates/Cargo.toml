[package]
name = "crates"
version = "0.0.0"
authors.workspace = true
edition.workspace = true
rust-version.workspace = true
homepage.workspace = true
repository.workspace = true
license.workspace = true
keywords.workspace = true
categories.workspace = true

[dev-dependencies]
insta.workspace = true

[build-dependencies]
cargo_metadata.workspace = true

[lints]
workspace = true
