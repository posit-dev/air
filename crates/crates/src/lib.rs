// Generates `AIR_CRATE_NAMES`, a const array of the crate names in the air workspace,
// see `crates/build.rs`
include!(concat!(env!("OUT_DIR"), "/lib.rs"));

#[cfg(test)]
mod tests {
    use crate::AIR_CRATE_NAMES;

    #[test]
    fn test_crate_names() {
        insta::assert_debug_snapshot!(AIR_CRATE_NAMES);
    }
}
