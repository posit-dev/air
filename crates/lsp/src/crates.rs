// Generates `AIR_CRATE_NAMES`, a const array of the crate names in the air workspace,
// see `lsp/src/build.rs`
include!(concat!(env!("OUT_DIR"), "/crates.rs"));

#[cfg(test)]
mod tests {
    use crate::crates::AIR_CRATE_NAMES;

    #[test]
    fn test_crate_names() {
        insta::assert_debug_snapshot!(AIR_CRATE_NAMES);
    }
}
