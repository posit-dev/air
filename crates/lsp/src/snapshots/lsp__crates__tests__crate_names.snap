---
source: crates/lsp/src/crates.rs
expression: AIR_CRATE_NAMES
---
[
    "air",
    "air_formatter_test",
    "air_r_factory",
    "air_r_formatter",
    "air_r_parser",
    "air_r_syntax",
    "biome_ungrammar",
    "fs",
    "line_ending",
    "lsp",
    "lsp_test",
    "tests_macros",
    "workspace",
    "xtask",
    "xtask_codegen",
]
